packages/react-components/theme-designer/src/components/ColorTokens/useColorOverrideReducer.ts

```diff
@@ -0,0 +1,104 @@
+import * as React from 'react';
+import { Brands, BrandVariants } from '@fluentui/react-theme';
+import type { AppState } from '../../useThemeDesignerReducer';
+
+export type ColorOverrideBrands = Record<string, Brands>;
+
+export type ColorOverrides = {
+  teamsLight: ColorOverrideBrands;
+  teamsDark: ColorOverrideBrands;
+  webLight: ColorOverrideBrands;
+  webDark: ColorOverrideBrands;
+  customLight: ColorOverrideBrands;
+  customDark: ColorOverrideBrands;
+};
+
+export type DispatchColorOverrides = {
+  type: string;
+  colorToken?: string;
+  newValue?: Brands;
+};
+
+export const getCurrentOverride = (appState: AppState, colorOverride: ColorOverrides) => {
+  switch (appState.themeLabel) {
+    case 'Teams Light':
+      return colorOverride.teamsLight;
+    case 'Teams Dark':
+      return colorOverride.teamsDark;
+    case 'Web Light':
+      return colorOverride.webLight;
+    case 'Web Dark':
+      return colorOverride.webDark;
+    case 'Custom':
+      if (!appState.isDark) {
+        return colorOverride.customLight;
+      } else {
+        return colorOverride.customDark;
+      }
+    default:
+      return colorOverride.customLight;
+  }
+};
+
+export const useColorOverrideReducer = (appState: AppState, brand: BrandVariants) => {
+  const colorOverrideReducer: (
```

Comment:
this reducer is pure now so you can hoist it outside of the hook to avoid recreating it on every invocation